File: src/scripts/load_data.py
import glob

import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score, root_mean_squared_error

from src.ncvreg.models.ncvreg import NCVREG

# Pick 4 random csv files from a directory and load then into 1 dataframe
df = pd.DataFrame()
files = glob.glob("/Users/jrp208/Documents/Independent_Work/data/kaggle_stocks/stocks/*.csv")

# Randomly select n files from file
idxs = np.random.randint(0, int(len(files)), 4,)
for idx in idxs:
    file = files[idx]
    temp = pd.read_csv(file)
    if temp.isnull().values.any():
        continue
    temp.columns = [x.lower().replace(" ", "_") for x in temp.columns]
    # Add stock name to temp
    temp['symbol'] = file.split('.')[0].split('/')[-1]
    df = pd.concat([df, temp], axis=0)

X = df.drop(columns=['date', 'symbol', 'adj_close']).to_numpy()
y = df['adj_close'].to_numpy()

X_train, X_test, y_train, y_test = train_test_split(X, y)

model = NCVREG(X_train, y_train)
model.fit()

res = model.predict(X_test)

print(root_mean_squared_error(y_test, res))
print(r2_score(y_test, res))

File: src/scripts/load_data.R
library(tidyverse)
source("/Users/jrp208/Documents/CWRU/Li_lab/ncvreg/src/R/ncvreg/R/ncvreg.R")

# Load data
# Pick 4 random csv files from a directory and load then into 1 dataframe
df <- data.frame()
files <- list.files(path = "/Users/jrp208/Documents/Independent_Work/data/kaggle_stocks/stocks", pattern = "\\.csv$", full.names = TRUE)

# Randomly select n files from file
idxs <- sample(1:length(files), 4, replace = FALSE)
for (idx in idxs) {
    file <- files[idx]
    temp <- read_csv(file)
    names(temp) <- tolower(names(temp))
    names(temp) <- str_replace_all(names(temp), " ", "_")
    # Add stock name to temp
    symbol <- str_split(str_split(file, "/")[-1], "\\.")[1]
    temp <- temp %>% mutate(symbol = symbol)

    df <- bind_rows(df, temp)
}

X <- df %>% select(-date, -symbol, -adj_close) %>% as.matrix()
y <- df$adj_close %>% as.matrix()

model <- ncvreg(X, y, penalty = "lasso")


